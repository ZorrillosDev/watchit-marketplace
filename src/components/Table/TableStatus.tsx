import styled from 'styled-components'
import { Typography, WithTheme, withTheme } from '@material-ui/core'
import React, { FC, PropsWithChildren } from 'react'

export const TableStatus: FC<PropsWithChildren<{ color: string }>> = (props): JSX.Element => {
  return (
    <TableStatusWrapper {...props}>
      <Typography>
        {props.children}
      </Typography>
    </TableStatusWrapper>
  )
}

export default TableStatus

export const TableStatusWrapper = withTheme(
  styled.div<WithTheme & { color: string }>`
    display: flex;
    padding: 6px 1rem;
    width: fit-content;
    align-items: center;
    justify-content: flex-start;
    border-radius: 5px;
    line-height: 1;
    text-transform: lowercase;
    transform: translateY(2px);
    background-color: ${({ theme, color }) => theme.palette[color].light};
    
    p {
      color: ${({ theme, color }) => theme.palette[color].dark} !important;
      font-weight: bold;
      text-transform: lowercase;
    }
  `
)
