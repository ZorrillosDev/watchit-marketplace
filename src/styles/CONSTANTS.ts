// Theme color constants
export const THEME_COLOR_PRIMARY_LIGHT = '#e3f2fd'
export const THEME_COLOR_PRIMARY_MAIN = '#2286A5'
export const THEME_COLOR_PRIMARY_DARK = '#155f75'

export const THEME_COLOR_SECONDARY_LIGHT = '#ede7f6'
export const THEME_COLOR_SECONDARY_MAIN = '#673ab7'
export const THEME_COLOR_SECONDARY_DARK = '#5e35b1'

export const THEME_COLOR_WARNING_LIGHT = '#fff8e1'
export const THEME_COLOR_WARNING_MAIN = '#ffe57f'
export const THEME_COLOR_WARNING_DARK = '#ffc107'

export const THEME_COLOR_SUCCESS_LIGHT = '#b9f6ca'
export const THEME_COLOR_SUCCESS_MAIN = '#20bf6b'
export const THEME_COLOR_SUCCESS_DARK = '#02883a'

export const THEME_COLOR_ERROR_LIGHT = '#ef9a9a'
export const THEME_COLOR_ERROR_MAIN = '#f44336'
export const THEME_COLOR_ERROR_DARK = '#c62828'

export const THEME_COLOR_GREY_50 = '#fafafa'
export const THEME_COLOR_GREY_100 = '#f5f5f5'
export const THEME_COLOR_GREY_200 = '#eeeeee'
export const THEME_COLOR_GREY_300 = '#e0e0e0'
export const THEME_COLOR_GREY_500 = '#9e9e9e'
export const THEME_COLOR_GREY_600 = '#757575'
export const THEME_COLOR_GREY_700 = '#616161'
export const THEME_COLOR_GREY_900 = '#212121'

export const THEME_BACKGROUND_DEFAULT = '#ffffff'
export const THEME_BACKGROUND_PAPER = '#ffffff'

export const THEME_SHAPE_BORDER_RADIUS: number = 8

export const THEME_GRID_SPACING = 2

export const THEME_DARK_COLOR_PRIMARY_LIGHT = '#29314f'
export const THEME_DARK_COLOR_PRIMARY_MAIN = '#2196f3'
export const THEME_DARK_COLOR_PRIMARY_DARK = '#1e88e5'

export const THEME_DARK_COLOR_SECONDARY_LIGHT = '#1b1e45'
export const THEME_DARK_COLOR_SECONDARY_MAIN = '#7c4dff'
export const THEME_DARK_COLOR_SECONDARY_DARK = '#651fff'

export const THEME_DARK_COLOR_WARNING_LIGHT = '#3a4058'
export const THEME_DARK_COLOR_WARNING_MAIN = '#ffe57f'
export const THEME_DARK_COLOR_WARNING_DARK = '#d6a62e'

export const THEME_DARK_COLOR_SUCCESS_LIGHT = '#3a4a5b'
export const THEME_DARK_COLOR_SUCCESS_MAIN = '#00e676'
export const THEME_DARK_COLOR_SUCCESS_DARK = '#17b956'

export const THEME_DARK_COLOR_ERROR_LIGHT = '#ef9a9a'
export const THEME_DARK_COLOR_ERROR_MAIN = '#f44336'
export const THEME_DARK_COLOR_ERROR_DARK = '#c62828'

export const THEME_DARK_COLOR_GREY_50 = '#fafafa'
export const THEME_DARK_COLOR_GREY_100 = '#f5f5f5'
export const THEME_DARK_COLOR_GREY_200 = '#bdc8f0'
export const THEME_DARK_COLOR_GREY_300 = '#353c5a'
export const THEME_DARK_COLOR_GREY_500 = '#8492c4'
export const THEME_DARK_COLOR_GREY_600 = '#757575'
export const THEME_DARK_COLOR_GREY_700 = '#bdc8f0'
export const THEME_DARK_COLOR_GREY_900 = '#d7dcec'

export const THEME_DARK_BACKGROUND_DEFAULT = '#111936'
export const THEME_DARK_BACKGROUND_PAPER = '#212946'
