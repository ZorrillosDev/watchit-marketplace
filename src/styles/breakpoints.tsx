import { hideAt } from '@styles/responsive'
import {
  RESPONSIVE_DESKTOP,
  RESPONSIVE_DESKTOP_L,
  RESPONSIVE_LAPTOP,
  RESPONSIVE_LAPTOP_L,
  RESPONSIVE_MOBILE_L,
  RESPONSIVE_MOBILE_M,
  RESPONSIVE_MOBILE_S,
  RESPONSIVE_TABLET
} from '@styles/CONSTANTS'

export const MobileSmallAndDown = hideAt({ min: RESPONSIVE_MOBILE_S })
export const MobileMediumAndDown = hideAt({ min: RESPONSIVE_MOBILE_M })
export const MobileLargeAndDown = hideAt({ min: RESPONSIVE_MOBILE_L })
export const MobileOnly = hideAt({ max: RESPONSIVE_MOBILE_S, min: RESPONSIVE_TABLET })
export const TabletAndDown = hideAt({ min: RESPONSIVE_TABLET })
export const TabletAndUp = hideAt({ max: RESPONSIVE_TABLET })
export const TabletOnly = hideAt({ max: RESPONSIVE_TABLET, min: RESPONSIVE_LAPTOP })
export const LaptopAndDown = hideAt({ min: RESPONSIVE_LAPTOP })
export const LaptopAndUp = hideAt({ max: RESPONSIVE_LAPTOP })
export const LaptopOnly = hideAt({ max: RESPONSIVE_LAPTOP, min: RESPONSIVE_LAPTOP_L })
export const LaptopLargeAndDown = hideAt({ min: RESPONSIVE_LAPTOP_L })
export const LaptopLargeAndUp = hideAt({ max: RESPONSIVE_LAPTOP_L })
export const LaptopLargeOnly = hideAt({ max: RESPONSIVE_LAPTOP_L, min: RESPONSIVE_DESKTOP })
export const DesktopAndDown = hideAt({ min: RESPONSIVE_DESKTOP })
export const DesktopAndUp = hideAt({ max: RESPONSIVE_DESKTOP })
export const DesktopOnly = hideAt({ max: RESPONSIVE_DESKTOP, min: RESPONSIVE_DESKTOP_L })
export const DesktopLargeAndUp = hideAt({ max: RESPONSIVE_DESKTOP_L })
