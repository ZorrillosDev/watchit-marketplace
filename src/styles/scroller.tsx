import { css } from 'styled-components'

export default css`
  margin-right: 0.5rem !important;
  
  &::-webkit-scrollbar {
    width: 0.5rem;
    height: 0.4rem;
    padding-right: 0.5rem;
    background-color: white;
    -webkit-transition: all 0.3s ease-in-out;
    -moz-transition: all 0.3s ease-in-out;
    transition: all 0.3s ease-in-out;
  }

  &::-webkit-scrollbar-button, &::-webkit-scrollbar-corner, &::-webkit-resizer {
    display: none;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: transparent;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: #999;
    border-radius: 0.25rem;
    -webkit-transition: all 0.3s ease-in-out;
    -moz-transition: all 0.3s ease-in-out;
    transition: all 0.3s ease-in-out;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: #666;
  }
`
