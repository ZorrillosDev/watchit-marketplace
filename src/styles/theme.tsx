import {
  THEME_COLOR_PRIMARY_CONTRAST,
  THEME_COLOR_PRIMARY_DARK,
  THEME_COLOR_PRIMARY_LIGHT,
  THEME_COLOR_PRIMARY_MAIN
} from '@styles/CONSTANTS'

// Global styles can be moved to a separate file for ease of maintenance.
export const globalOverrides = {
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          margin: 0,
          padding: 0,
          height: '100%'
        },
        body: {
          margin: 0,
          padding: 0,
          height: '100%',
          fontSize: '1rem',
          backgroundColor: '#fff'
        }
      }
    }
  }
}

export const typography = {
  typography: {
    fontFamily: [
      'Tamil Sangam MN',
      'Oswald',
      'Arial',
      'sans-serif'
    ].join(','),
    h3: {
      fontWeight: 700,
      lineHeight: 1,
      '@media (min-width: 1900px)': {
        fontSize: '3.5rem'
      }
    },
    h5: {
      fontWeight: 600,
      lineHeight: 1,
      '@media (min-width: 1900px)': {
        fontSize: '2rem'
      }
    },
    h6: {
      fontWeight: 600,
      lineHeight: 1,
      '@media (min-width: 1900px)': {
        fontSize: '2rem'
      }
    },
    body1: {
      lineHeight: 1,
      '@media (min-width: 1900px)': {
        fontSize: '1.3rem'
      }
    },
    body2: {
      lineHeight: 1,
      '@media (min-width: 1900px)': {
        fontSize: '1.2rem'
      }
    },
    button: {
      fontSize: '1rem',
      fontWeight: 700
    }
  }
}

export const defaultTheme = {
  palette: {
    primary: {
      light: THEME_COLOR_PRIMARY_LIGHT,
      main: THEME_COLOR_PRIMARY_MAIN,
      dark: THEME_COLOR_PRIMARY_DARK,
      contrastText: THEME_COLOR_PRIMARY_CONTRAST
    }
  }
}
