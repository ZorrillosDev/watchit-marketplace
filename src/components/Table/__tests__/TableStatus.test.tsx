import TableStatus, { TableStatusWrapper } from '@components/Table/TableStatus'
import { Theme, Typography } from '@material-ui/core'
import { mount, shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { defaultTheme } from '@styles/theme'

describe('<TableStatus />', () => {
  let theme: Theme

  beforeAll(() => {
    theme = createTheme(defaultTheme)
  })

  it('should render table status', () => {
    const component = shallow(<TableStatus color='success' />)
    expect(component).toMatchSnapshot()
  })

  it('should render table status processing status', () => {
    const component = shallow(
      <TableStatus color='warning'>
        Processing
      </TableStatus>)
    const text = component.find(Typography)

    expect(text.text()).toEqual('Processing')
  })

  it('should render table status sent status', () => {
    const component = shallow(
      <TableStatus color='warning'>
        Sent
      </TableStatus>)
    const text = component.find(Typography)

    expect(text.text()).toEqual('Sent')
  })

  it('should have table status wrapper processing styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <TableStatusWrapper color='warning' />
      </ThemeProvider>
    )

    expect(component).toHaveStyleRule('background-color', '#ffb74d')
    expect(component).toHaveStyleRule('background-color', theme.palette.warning.light)
  })

  it('should have table status wrapper sent styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <TableStatusWrapper color='success' />
      </ThemeProvider>
    )

    expect(component).toHaveStyleRule('background-color', '#81c784')
    expect(component).toHaveStyleRule('background-color', theme.palette.success.light)
  })
})
