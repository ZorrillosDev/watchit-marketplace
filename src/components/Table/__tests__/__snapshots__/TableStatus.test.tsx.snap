// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableStatus /> should render table status 1`] = `ShallowWrapper {}`;
